src/store/attachment.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * @copyright Copyright (c) 2020 Jakob Röhrl <jakob.roehrl@web.de>
+ *
+ * @author Jakob Röhrl <jakob.roehrl@web.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { AttachmentApi } from './../services/AttachmentApi'
+import Vue from 'vue'
+
+const apiClient = new AttachmentApi()
+
+export default {
+	state: {
+		attachments: {},
+	},
+	getters: {
+		attachmentsByCard: state => (cardId) => {
+			if (typeof state.attachments[cardId] === 'undefined') {
+				return []
+			}
+			return state.attachments[cardId]
+		},
+
+	},
+	mutations: {
+		createAttachment(state, { cardId, attachments }) {
```

Comment:
Naming it attachment would be more obvious since it is just about one single attachment.

Reply:
This mutation is called to generate all attachment entries on card load, too.

Reply:
Then we should split the mutation into two I'd say. We could get into some weired state with the current implementation if you call createAttachment for some reason with an array of attachments twice.